Automattic/wp-calypso
# Domain Management i1: Move function to process domain status into domains-table packages

This PR involves huge changes that moves a lot of parts that's required to resolve the domain status into the domains-table packages and the data-sources package. 

There are certain branches in the `resolveDomainStatus` function that has been removed as they are not used in the codebase.

Bear in mind also that as we move these parts to a central location additional refactoring will be needed to cleanup other areas with the same code so that we will have one source of truth.


Closes https://github.com/Automattic/dotcom-forge/issues/3423

## Proposed Changes

* Move the `ResponseDomain` type and it's related dependencies into `@automattic/data-sources ` package so that they can be accessed in the domains table and elsewhere.
* Move `resolveDomainStatus` method to the domains-table package along with util functions and dependencies. This allow us to resolve the domain status and can also add sorting to the status column.
* Actions relating to purchases are passed as props to remove the direct dependency in the domains table.

No doubt we will need some tests and more clean-up but this PR is already so big.

## Testing Instructions



* go to `domains/manage?flags=domains/management`
* Verify the status displayed are the same as the ones at `/domains/manage`

## Pre-merge Checklist
